xscen/diagnostics.py

```diff
@@ -97,535 +110,256 @@ def _invert_unphysical_temperatures(
-
-    vmin = min(
-        min(sim.quantile(0.05), scen.quantile(0.05)),
-        -max(sim.quantile(0.95), scen.quantile(0.95)),
-    )
 
-    ps = sim.plot(
-        ax=axsm,
-        vmin=vmin,
-        vmax=-vmin,
-        cmap=cmbias,
-        add_colorbar=False,
-        transform=crs.PlateCarree(),
-    )
-    scen.plot(
-        ax=axsc,
-        vmin=vmin,
-        vmax=-vmin,
-        cmap=cmbias,
-        add_colorbar=False,
-        transform=crs.PlateCarree(),
-    )
-    if op == "distance":
-        diff = abs(sim - scen)
-    elif op == "improvement":
-        diff = abs(sim) - abs(scen)
-    elif op == "difference":  # op == 'diff':
-        diff = sim - scen
+    if isinstance(properties, (str, Path)):
+        logger.debug("Loading properties module.")
+        module = load_xclim_module(properties)
+        properties = module.iter_indicators()
+    elif hasattr(properties, "iter_indicators"):
+        properties = properties.iter_indicators()
+
+    try:
+        N = len(properties)
+    except TypeError:
+        N = None
     else:
-        raise NotImplementedError(
-            "Operation not implemented. Try 'distance', 'improvement' or 'difference' instead."
+        logger.info(f"Computing {N} properties.")
+
+    # select periods for ds
+    if period is not None and "time" in ds:
+        ds = ds.sel({"time": slice(str(period[0]), str(period[1]))})
+        date_start = ds.time.values[0]
+        date_end = ds.time.values[-1]
+    # select periods for ref_measure
+    if (
+        dref_for_measure is not None
+        and period is not None
+        and "time" in dref_for_measure
+    ):
+        dref_for_measure = dref_for_measure.sel(
+            {"time": slice(str(period[0]), str(period[1]))}
         )
-    pc = diff.plot(
-        ax=axim,
-        robust=True,
-        cmap=cmimpr,
-        center=0,
-        add_colorbar=False,
-        transform=crs.PlateCarree(),
-    )
 
-    fig.suptitle(title, fontsize="x-large")
-    axsm.set_title(sim.name.replace("_", " ").capitalize())
-    axsc.set_title(scen.name.replace("_", " ").capitalize())
-    axim.set_title(op.capitalize())
-    fig.tight_layout()
-    fig.colorbar(
-        ps,
-        cax=plt.subplot(gs[1, 0]),
-        shrink=0.5,
-        label=sim.attrs.get("long_name", sim.name),
-        orientation="horizontal",
+    if unstack:
+        ds = unstack_fill_nan(ds)
+
+    unit_conversion = unit_conversion or {}
+    for var, unit in unit_conversion.items():
+        ds[var] = convert_units_to(ds[var], unit)
+
+    prop = xr.Dataset()  # dataset with all properties
+    meas = xr.Dataset()  # dataset with all measures
+    for i, ind in enumerate(properties, 1):
+        if isinstance(ind, tuple):
+            iden, ind = ind
+        else:
+            iden = ind.identifier
+        logger.info(f"{i} - Computing {iden}.")
+
+        # Make the call to xclim
+        prop[iden] = ind(ds=ds)
```

Comment:
Ceci empêche l'utilisation de `var_name` dans l'indicateur. Ce champ peut contenir des balises et donc être formaté selon les arguments d'entrée, au contraire de  `identifier`. C'est sûr que ça n'arrive pas souvent avec des modules "custom", mais si jamais?
```suggestion
        out = ind(ds=ds)
        vname = out.name
        prop[vname] = out
```
et utiliser `vname` là où approprié plus bas.